.github/workflows/build.yml

```diff
@@ -51,7 +51,12 @@ jobs:
       - name: Build ckan.exe and netkan.exe
         run: ./build --configuration=${{ matrix.configuration}}
       - name: Run tests
-        run: xvfb-run ./build test+only --configuration=${{ matrix.configuration }} --where="Category!=FlakyNetwork"
+        run: |
+          xvfb-run ./build test+only --configuration=${{ matrix.configuration }} --where="Category!=FlakyNetwork"
+          # Smoke test for netkan.exe, test for SpaceDock reachability
+          cd _build && \
+            curl -O https://raw.githubusercontent.com/KSP-CKAN/NetKAN/master/NetKAN/ZeroMiniAVC.netkan && \
+            mono netkan.exe ZeroMiniAVC.netkan
```

Comment:
Will this test the container? I thought it would have to be added to the Dockerfiles.

Reply:
The workflow runs in mono containers, so it is mostly the same.
The problem is that we don't build the images in `build.yml`, only `deploy.yml`.

We could also build the images in `build.yml` (manually, not using `./build docker-*`), but we would also need to split the Dockerfile into a multi-stage file so we don't bloat the production images with unrelated files.
This is more work than I'd like to include in this PR, I'd prefer to get our infrastructure working again first.

Btw., I just found out that if we remove the `--where="Category!=FlakyNetwork"` we also reproduce today's problem:
```
1) Failed : Tests.Core.Net.Net.SpaceDockSSL
Expected: No Exception to be thrown
  But was:  <System.Net.WebException: Error: TrustFailure (Authentication failed, see inner exception.) ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception. ---> Mono.Btls.MonoBtlsException: Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED
  
2) Error : Tests.NetKAN.Sources.Jenkins.JenkinsApiTests.GetLatestBuild_ModuleManager_Works
System.Net.WebException : Error: TrustFailure (Authentication failed, see inner exception.)
  ----> System.Security.Authentication.AuthenticationException : Authentication failed, see inner exception.
  ----> Mono.Btls.MonoBtlsException : Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED

3) Error : Tests.NetKAN.Sources.Spacedock.SpacedockApiTests.GetsModCorrectly
System.Net.WebException : Error: TrustFailure (Authentication failed, see inner exception.)
  ----> System.Security.Authentication.AuthenticationException : Authentication failed, see inner exception.
  ----> Mono.Btls.MonoBtlsException : Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED

4) Failed : Tests.NetKAN.Sources.Spacedock.SpacedockApiTests.ThrowsWhenModMissing
  Expected: instance of <CKAN.Kraken>
  But was:  <System.Net.WebException: Error: TrustFailure (Authentication failed, see inner exception.) ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception. ---> Mono.Btls.MonoBtlsException: Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED
```

However I'm not sure if we want to run so many network tests in our CI, for one everything breaks if SpaceDock is down, and it generates load for it as well.

Reply:
> The workflow runs in mono containers, so it is mostly the same.

Except now _all_ pull requests will fail the tests, right? We need something that will succeed after this PR's changes.

Reply:
Latest changes should do it the way you have in mind. It's ugly but it works...
...on those Docker images that run on a Debian release that is still supported by Docker.

I can move it into a separate step and let it only run for certain mono versions. I could choose all supported ones, or maybe just `latest`?

Reply:
(Btw., if you have a better test candidate than ZeroMiniAVC, suggestions welcome, that was just the one I spotted at the end of the message flood in #netkan-bot earlier)

Reply:
This makes sense, thanks!